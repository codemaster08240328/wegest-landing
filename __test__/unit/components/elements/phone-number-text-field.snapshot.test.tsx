import React from 'react'
import renderer from 'react-test-renderer'
import PhoneNumberTextField from '@/components/elements/phone-number-text-field'

describe('<PhoneNumberTextField> snapshot', () => {
  it('renders correctly when there are no props', () => {
    expect(renderer.create(<PhoneNumberTextField />).toJSON()).toMatchSnapshot()
  })

  it('renders correctly when defaultCountry is passed', () => {
    expect(
      renderer.create(<PhoneNumberTextField defaultCountry='se' />).toJSON(),
    ).toMatchSnapshot()
  })

  it('renders correctly when there is label', () => {
    expect(
      renderer.create(<PhoneNumberTextField label='Phone' />).toJSON(),
    ).toMatchSnapshot()
  })

  it('renders correctly when there is className', () => {
    expect(
      renderer.create(<PhoneNumberTextField className='class-1' />).toJSON(),
    ).toMatchSnapshot()
  })

  it('renders correctly when there is value', () => {
    expect(
      renderer.create(<PhoneNumberTextField value='123454578' />).toJSON(),
    ).toMatchSnapshot()
  })

  it('renders correctly when there is helper text', () => {
    expect(
      renderer
        .create(<PhoneNumberTextField helperText='Some important text' />)
        .toJSON(),
    ).toMatchSnapshot()
  })
})
