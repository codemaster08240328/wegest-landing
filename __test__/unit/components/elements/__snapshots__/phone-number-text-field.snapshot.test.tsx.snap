// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PhoneNumberTextField> snapshot renders correctly when defaultCountry is passed 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root makeStyles-root-1"
  data-testid="class"
  onClick={[Function]}
  onKeyDown={[Function]}
>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedStart"
    onClick={[Function]}
  >
    <div
      className="MuiInputAdornment-root MuiPhoneNumber-positionStart MuiInputAdornment-positionStart"
    >
      <button
        aria-haspopup={true}
        aria-label="Select country"
        aria-owns={null}
        className="MuiButtonBase-root MuiButton-root MuiButton-text MuiPhoneNumber-flagButton"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <div
            className="flag se MuiPhoneNumber-flag"
          />
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedStart"
      disabled={false}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="+1 (702) 123-4567"
      required={false}
      type="tel"
      value="+46"
    />
  </div>
</div>
`;

exports[`<PhoneNumberTextField> snapshot renders correctly when there are no props 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root makeStyles-root-1"
  data-testid="class"
  onClick={[Function]}
  onKeyDown={[Function]}
>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedStart"
    onClick={[Function]}
  >
    <div
      className="MuiInputAdornment-root MuiPhoneNumber-positionStart MuiInputAdornment-positionStart"
    >
      <button
        aria-haspopup={true}
        aria-label="Select country"
        aria-owns={null}
        className="MuiButtonBase-root MuiButton-root MuiButton-text MuiPhoneNumber-flagButton"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <div
            className="flag it MuiPhoneNumber-flag"
          />
        </span>
      </button>
    </div>
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedStart"
      disabled={false}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="+1 (702) 123-4567"
      required={false}
      type="tel"
      value="+39"
    />
  </div>
</div>
`;

exports[`<PhoneNumberTextField> snapshot renders correctly when there is className 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root makeStyles-root-1 class-1"
  data-testid="class"
  onClick={[Function]}
  onKeyDown={[Function]}
>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedStart"
    onClick={[Function]}
  >
    <div
      className="MuiInputAdornment-root MuiPhoneNumber-positionStart MuiInputAdornment-positionStart"
    >
      <button
        aria-haspopup={true}
        aria-label="Select country"
        aria-owns={null}
        className="MuiButtonBase-root MuiButton-root MuiButton-text MuiPhoneNumber-flagButton"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <div
            className="flag it MuiPhoneNumber-flag"
          />
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedStart"
      disabled={false}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="+1 (702) 123-4567"
      required={false}
      type="tel"
      value="+39"
    />
  </div>
</div>
`;

exports[`<PhoneNumberTextField> snapshot renders correctly when there is helper text 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root makeStyles-root-1"
  data-testid="class"
  onClick={[Function]}
  onKeyDown={[Function]}
>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedStart"
    onClick={[Function]}
  >
    <div
      className="MuiInputAdornment-root MuiPhoneNumber-positionStart MuiInputAdornment-positionStart"
    >
      <button
        aria-haspopup={true}
        aria-label="Select country"
        aria-owns={null}
        className="MuiButtonBase-root MuiButton-root MuiButton-text MuiPhoneNumber-flagButton"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <div
            className="flag it MuiPhoneNumber-flag"
          />
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedStart"
      disabled={false}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="+1 (702) 123-4567"
      required={false}
      type="tel"
      value="+39"
    />
  </div>
  <p
    className="MuiFormHelperText-root"
  >
    Some important text
  </p>
</div>
`;

exports[`<PhoneNumberTextField> snapshot renders correctly when there is label 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root makeStyles-root-1"
  data-testid="class"
  onClick={[Function]}
  onKeyDown={[Function]}
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
    data-shrink={true}
  >
    Phone
  </label>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedStart"
    onClick={[Function]}
  >
    <div
      className="MuiInputAdornment-root MuiPhoneNumber-positionStart MuiInputAdornment-positionStart"
    >
      <button
        aria-haspopup={true}
        aria-label="Select country"
        aria-owns={null}
        className="MuiButtonBase-root MuiButton-root MuiButton-text MuiPhoneNumber-flagButton"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <div
            className="flag it MuiPhoneNumber-flag"
          />
        </span>
      </button>
    </div>
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedStart"
      disabled={false}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="+1 (702) 123-4567"
      required={false}
      type="tel"
      value="+39"
    />
  </div>
</div>
`;

exports[`<PhoneNumberTextField> snapshot renders correctly when there is value 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root makeStyles-root-1"
  data-testid="class"
  onClick={[Function]}
  onKeyDown={[Function]}
>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedStart"
    onClick={[Function]}
  >
    <div
      className="MuiInputAdornment-root MuiPhoneNumber-positionStart MuiInputAdornment-positionStart"
    >
      <button
        aria-haspopup={true}
        aria-label="Select country"
        aria-owns={null}
        className="MuiButtonBase-root MuiButton-root MuiButton-text MuiPhoneNumber-flagButton"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <div
            className="flag it MuiPhoneNumber-flag"
          />
        </span>
      </button>
    </div>
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedStart"
      disabled={false}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="+1 (702) 123-4567"
      required={false}
      type="tel"
      value="+12 345 4578"
    />
  </div>
</div>
`;
